sunpy/lightcurve/sources/goes.py

```diff
@@ -74,10 +75,13 @@ def peek(self, title="GOES Xray Flux"):
     @classmethod
     def _get_default_uri(cls):
         """Retrieve latest GOES data if no other data is specified"""
-        today = datetime.datetime.today()
-        days_back = 3
-        time_range = TimeRange(today - datetime.timedelta(days=days_back),
-                               today - datetime.timedelta(days=days_back - 1))
+        now = datetime.datetime.utcnow()
+        time_range = TimeRange(datetime.datetime(now.year, now.month, now.day), now)
+        url_does_exist = net.url_exists(cls._get_url_for_date_range(time_range))
+        while not url_does_exist:
+            time_range = TimeRange(time_range.start-datetime.timedelta(days=1),
+                                   time_range.start)
+            url_does_exist = net.url_exists(cls._get_url_for_date_range(time_range))
```

Comment:
Previously this was trying to provide a 3 days lightcurve (probably as the standard given by NOAA plots), now it will give just the last day available.  Is this what we want?